crates/cheatcodes/src/test/expect.rs

```diff
@@ -436,9 +436,12 @@ pub(crate) fn handle_expect_emit(state: &mut Cheatcodes, log: &alloy_primitives:
 
     if expected_topic_0
         .zip(log_topic_0)
-        .map_or(false, |(a, b)| a == b && expected.topics().len() == log.topics().len())
+        .map_or(false, |(a, b)| a == b && expected.topics().len() == log.topics().len()) ||
+        // Anonymous events are a special case where topic 0 is not generated.
+        // If the anonymous event has no other topics we will still attempt to match the source address and data if requested.
+        (expected_topic_0.is_none() && log_topic_0.is_none())
```

Comment:
I don't understand this check,

this shouldn't be reachable if any of the topics is none, because we zip them?

Reply:
It did work before but I've modified the behaviour to use the `.map_or` default case to evaluate the anonymous event case. This is evaluated when zipped to `None`.